### PR TITLE
Add some more types

### PR DESCRIPTION
**Pull request by jenshnielsen**
This includes typing of all function in the gui and plot module

### CODE REVIEW

File: plottr/apps/inspectr.py [hunk 1]
```diff
@@ -136,7 +141,7 @@ def __init__(self, parent=None):
         self.itemSelectionChanged.connect(self.selectRun)
         self.itemActivated.connect(self.activateRun)
 
-    def addRun(self, runId, **vals):
+    def addRun(self, runId: int, **vals: str) -> None:
```
Comment (astafan8) [inline plottr/apps/inspectr.py:144]: type of `vals` being `str` here is for the "values" of the `**vals` dictionary, right?

File: plottr/apps/inspectr.py [hunk 2]
```diff
@@ -149,7 +154,7 @@ def addRun(self, runId, **vals):
         item = SortableTreeWidgetItem(lst)
         self.addTopLevelItem(item)
 
-    def setRuns(self, selection):
+    def setRuns(self, selection: Dict[int, Dict[str,str]]) -> None:
```
Comment (astafan8) [inline plottr/apps/inspectr.py]:
```suggestion
    def setRuns(self, selection: Dict[int, Dict[str, str]]) -> None:
```

File: plottr/apps/inspectr.py [hunk 3]
```diff
@@ -185,14 +190,14 @@ class RunInfo(QtWidgets.QTreeWidget):
...
-    def __init__(self, parent=None):
+    def __init__(self, parent: Optional[QtWidgets.QWidget] = None):
         super().__init__(parent)
 
         self.setHeaderLabels(['Key', 'Value'])
         self.setColumnCount(2)
 
     @Slot(dict)
-    def setInfo(self, infoDict):
+    def setInfo(self, infoDict: dict) -> None:
```
Comment (astafan8) [inline plottr/apps/inspectr.py]: as per signature of `dictToTreeWidgetItems` ?
```suggestion
    def setInfo(self, infoDict: Dict[str, Union[dict, str]]) -> None:
```

File: plottr/apps/inspectr.py [hunk 4]
```diff
@@ -477,12 +488,17 @@ def plotRun(self, runId):
         win.showTime()
 
 
-def inspectr(dbPath: str = None):
+class WindowDict(TypedDict):
+    flowchart: Flowchart
+    window: QCAutoPlotMainWindow
+
+
+def inspectr(dbPath: str = None) -> QtWidgets.QMainWindow:
```
Comment (astafan8) [inline plottr/apps/inspectr.py]: shouldn't we use QCodesDBInspector to be more specific? or is the idea that outside of this function the code only should care whether it is `QtWidgets.QMainWindow` or not?
```suggestion
def inspectr(dbPath: str = None) -> QCodesDBInspector:
```

File: plottr/gui/data_display.py [hunk 5]
```diff
@@ -47,10 +47,10 @@ def _makeItem(self, name):
         ])
 
     @Slot(int)
-    def _processCbChange(self, _):
+    def _processCbChange(self, _: Any) -> None:
```
Comment (astafan8) [inline plottr/gui/data_display.py]: according to the slot type it should be an integer
```suggestion
    def _processCbChange(self, _: int) -> None:
```

File: plottr/gui/widgets.py [hunk 6]
```diff
@@ -73,7 +73,8 @@ class PlotWindow(QtWidgets.QMainWindow):
 
     plotWidgetClass = MPLAutoPlot
 
-    def __init__(self, parent=None, fc: Flowchart = None, **kw):
+    def __init__(self, parent: Optional[QtWidgets.QMainWindow] = None,
+                 fc: Flowchart = None, **kw: Any):
```
Comment (astafan8) [inline plottr/gui/widgets.py]:
```suggestion
                 fc: Optional[Flowchart] = None, **kw: Any):
```

File: plottr/plot/base.py [hunk 7]
```diff
@@ -116,12 +116,12 @@ class PlotWidget(QtWidgets.QWidget):
     Implement a child class for actual plotting.
     """
 
-    def __init__(self, parent=None):
+    def __init__(self, parent: QtWidgets.QWidget = None) -> None:
```
Comment (astafan8) [inline plottr/plot/base.py]:
```suggestion
    def __init__(self, parent: Optional[QtWidgets.QWidget] = None) -> None:
```

**Review by astafan8 — APPROVED** [review]

File: plottr/apps/inspectr.py (hunk 1 above)
Comment (jenshnielsen) [reply plottr/apps/inspectr.py:144]: Yes that is how `**` is typed

File: plottr/gui/data_display.py (hunk 5 above)
Comment (jenshnielsen) [reply plottr/gui/data_display.py]: Yes but it's also unused so ?

Comment (astafan8) [reply plottr/gui/data_display.py]: yeah, unused inside the function, but from the outside i think the type should still be correct. in this case the "outside" is qt machinery of signals and slots, so it's not that important

**Pull request merged**